File: example_service_pkg/scripts/service_server.py
#!/usr/bin/env python3

import rospy
from example_service_pkg.srv import examplesrv, examplesrvResponse

def service_request_callback(req):
    rospy.loginfo("Request from client: [%s]", req.in_)
    response = "Service request received"
    rospy.loginfo("Response to client: [%s]", response)
    return examplesrvResponse(response)

def service_server():
    rospy.init_node('service_server_node')
    s = rospy.Service('test_service', examplesrv, service_request_callback)
    rospy.loginfo("Service server is ready to receive requests from clients.")
    rospy.spin()

if __name__ == "__main__":
    service_server()

File: example_service_pkg/src/service_server.cpp
#include "ros/ros.h"
#include "example_service_pkg/examplesrv.h"
#include <sstream>

bool service_callback(example_service_pkg::examplesrv::Request &req,
                      example_service_pkg::examplesrv::Response &res) {
    std::stringstream ss;
    ss << "Service request received";
    res.out = ss.str();
    ROS_INFO("Request from client: [%s]", req.in.c_str());
    ROS_INFO("Response to client: [%s]", res.out.c_str());
    return true;
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "service_server_node");
    ros::NodeHandle n;
    ros::ServiceServer service = n.advertiseService("test_service", service_callback);
    ROS_INFO("Service server is ready to receive requests from clients.");
    ros::spin();
    return 0;
}

